controllers/openstackcontrolplane_controller.go

```diff
@@ -103,7 +103,32 @@ func (r *OpenStackControlPlaneReconciler) Reconcile(ctx context.Context, req ctr
 	// Used in comparisons below to determine whether a status update is actually needed
 	newProvStatus := ospdirectorv1beta1.OpenStackControlPlaneProvisioningStatus{}
 
+	// ConfigMap - containing Tripleo Passwords
 	envVars := make(map[string]common.EnvSetter)
+	cmLabels := common.GetLabels(instance, controlplane.AppLabel, map[string]string{})
+
+	templateParameters := make(map[string]interface{})
+	templateParameters["TripleoPasswords"] = common.GeneratePasswords()
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	cm := []common.Template{
+		{
+			Name:           "tripleo-passwords",
```

Comment:
is there a reason why we create tripleo-passwords as a CM instead of a secret?

Reply:
You beat me to this. I'm going to move it to a Secret. This is quicker to just "throw in" here as we have the template stuff working with CMs. (I don't think we have equivalent functionality for Secrets).

Reply:
it should just work if you use common.EnsureSecrets [1] instead of common.EnsureConfigMaps in L128

[1] https://github.com/openstack-k8s-operators/osp-director-operator/blob/master/pkg/common/secret.go#L219

Reply:
sounds good. I didn't know we had that (or forgot)